remotedata_android/src/test/java/com/mercari/remotedata/android/ParcelableRemoteDataTest.kt

```diff
@@ -0,0 +1,154 @@
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Loading<String> = RemoteData.Loading(progress = 50, totalUnits = 100)
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Loading<String> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored shouldBeEqualTo value
+    }
+
+    @Test
+    fun putAndGetLoading_parcelables() {
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Loading<SampleParcelable> = RemoteData.Loading(progress = 50, totalUnits = 100)
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Loading<SampleParcelable> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored shouldBeEqualTo value
+    }
+
+    @Test
+    fun putAndGetSuccess_primitives() {
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Success<Int> = RemoteData.Success(100)
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Success<Int> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored shouldBeEqualTo value
+    }
+
+    @Test
+    fun putAndGetSuccess_strings() {
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Success<String> = RemoteData.Success("aaa")
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Success<String> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored shouldBeEqualTo value
+    }
+
+    @Test
+    fun putAndGetSuccess_parcelables() {
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Success<SampleParcelable> = RemoteData.Success(SampleParcelable("aaa"))
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Success<SampleParcelable> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored shouldBeEqualTo value
+    }
+
+    @Test
+    fun putAndGetFailure() {
+        // Given: a value to be stored in the Bundle
+        val value: RemoteData.Failure<ErrorKind> = RemoteData.Failure(NetworkError())
+        val marshalled: ByteArray = marshall(value)
+
+        // When: Store it to the Bundle
+        val restored: RemoteData.Failure<ErrorKind> = unmarshall(marshalled)
+
+        // Then: Restored value should be the same as the initial value
+        restored.error.shouldBeInstanceOf<NetworkError>()
+    }
+
+    @Test(expected = RuntimeException::class)
```

Comment:
[super nits] I think this is okay by itself, but I find the `assertThrows` to be much simpler to use.